src/infrastructure/NodeService.js

```diff
@@ -342,6 +342,45 @@ class NodeService {
 			throw Error('nodeRewardsController endpoint is not provided');
 	}
 
+	static getEnrollmentList = async (pageInfo, filterVaule, signerPublicKey) => {
+		const endpoint = globalConfig.endpoints.nodeRewardsController;
+		const { pageNumber, pageSize } = pageInfo;
+		const searchCriteria = {
+			pageNumber,
+			pageSize,
+			order: symbol.Order.Desc,
+			...filterVaule
+		};
+
+		let url = `${endpoint}/enrollments`;
+
+		if (signerPublicKey)
+			url += `/signerPublicKey/${signerPublicKey}`;
+
+		const response = (await Axios.get(url, { params: searchCriteria })).data;
+
+		const data = {
+			...response,
+			...response.pagination,
+			isLastPage: response.data.length < pageSize
```

Comment:
That is a condition if that is the last page.
and assume `pageSize` is 20, and `response.data.length` exactly 20.

so `isLastPage` should be `true` , but `response.data.length < pageSize` return `false` am I right?

Reply:
Yes, you are right. I've also added a check to the Pagination class. So if reqested page is empty, it will stay on previous page.
Otherwise we need solution which we used before - make a double reqest (pageNumber, pageNumber+1)